### ISSUE TITLE
No such file or directory

### ISSUE THREAD

**Issue by wisemonkey** [jensstein/oandbackup#108]
On marshmallow rooted and busy box installed,
Every time I try to back up app complains "no such file or directory".

**Comment by jensstein** [comment]
is this when backing up data or apks?
could you see if there is anything in the logfile (oandbackup.log) - i'm interested in which file or directory it cannot find.

**Comment by wisemonkey** [comment]
Here are messages from log file, I tried backing up both, followed by apk and data.



2015/11/22 - 00:13:18: cp: /mnt/shell/emulated/0/oandbackups/am.ed.exportcontacts: No such file or directory [Export Contacts]
2015/11/22 - 00:13:24: cp: /mnt/shell/emulated/0/oandbackups/am.ed.exportcontacts: No such file or directory [Export Contacts]
2015/11/22 - 00:13:30: cp: /mnt/shell/emulated/0/oandbackups/am.ed.exportcontacts: No such file or directory [Export Contacts]

On November 21, 2015 11:44:04 PM PST, jensstein <notifications@github.com> wrote:
>is this when backing up data or apks?
>could you see if there is anything in the logfile (oandbackup.log) -
>i'm interested in which file or directory it cannot find.
>
>---
>Reply to this email directly or view it on GitHub:
>https://github.com/jensstein/oandbackup/issues/108#issuecomment-158735584

-- 
Sent from my Android device with K-9 Mail. Please excuse my brevity.

**Comment by jensstein** [comment]
this is probably because the path to storage has changed. i do a very crude swap in ShellCommands.java:swapBackupDirPath (l. 1168) that should be smarter.
could i get you to run a backup with this apk and send me a logcat: https://drive.google.com/file/d/0B0bNUNWd2wPPVVJ6SWd5UV9Rcms/view?usp=sharing
(it's completely untested, since i don't have a marshmallow device and my emulator isn't useable for this right now - so this mainly for logging, the attempted fix will most likely fail).
you could also look around /mnt and /storage with a terminal emulator as the root user. the point is to find the path where the root user can access and write to the primary storage.

**Comment by wisemonkey** [comment]
Here is a path from terminal as a root
/storage/emulated/0/oandbackups

I'll try apk and let you know in a bit.

Just to make sure you need logcat not oandbackup log, correct?


On November 23, 2015 12:56:22 PM PST, jensstein <notifications@github.com> wrote:
>this is probably because the path to storage has changed. i do a very
>crude swap in ShellCommands.java:swapBackupDirPath (l. 1168) that
>should be smarter.
>could i get you to run a backup with this apk and send me a logcat:
>https://drive.google.com/file/d/0B0bNUNWd2wPPVVJ6SWd5UV9Rcms/view?usp=sharing
>(it's completely untested, since i don't have a marshmallow device and
>my emulator isn't useable for this right now - so this mainly for
>logging, the attempted fix will most likely fail).
>you could also look around /mnt and /storage with a terminal emulator
>as the root user. the point is to find the path where the root user can
>access and write to the primary storage.
>
>---
>Reply to this email directly or view it on GitHub:
>https://github.com/jensstein/oandbackup/issues/108#issuecomment-159061731

-- 
Sent from my Android device with K-9 Mail. Please excuse my brevity.

**Comment by jensstein** [comment]
yes, i need a logcat. and actually, you could also try this apk afterwards: https://drive.google.com/file/d/0B0bNUNWd2wPPVEVlN0tuakpHV28/view?usp=sharing
i don't need any logcat for that one, just see if it works.


also, just for the record of the thread, i accidentally deleted my comment above so here is what it more or less said:
this is most likely due to changes to the path to storage. i do a very crude swap in ShellCommands.java:swapBackupDirPath (l. 1168). it should be smarter.
please do a backup with this apk and send the logcat: https://drive.google.com/file/d/0B0bNUNWd2wPPVVJ6SWd5UV9Rcms/view?usp=sharing
(it's completely untested since i don't have a marshmallow device and my emulator isn't useable for this right now - so this is just for logging, the attempted fix will most likely fail).
you could also try looking around /mnt and /storage with a terminal emulator as the root user. the point is to find the path to storage which the root user can access and write to.

**Comment by wisemonkey** [comment]
Not sure if attachment works with github reply, but I've attached the logcat for oandbackup let me view if its good

On November 23, 2015 1:06:44 PM PST, jensstein <notifications@github.com> wrote:
>yes, i need a logcat. and actually, you could also try this apk
>afterwards:
>https://drive.google.com/file/d/0B0bNUNWd2wPPVEVlN0tuakpHV28/view?usp=sharing
>i don't need any logcat for that one, just see if it works.
>
>
>also, just for the record of the thread, i accidentally deleted my
>comment above so here is what it more or less said:
>this is most likely due to changes to the path to storage. i do a very
>crude swap in ShellCommands.java:swapBackupDirPath (l. 1168). it should
>be smarter.
>please do a backup with this apk and send the logcat:
>https://drive.google.com/file/d/0B0bNUNWd2wPPVVJ6SWd5UV9Rcms/view?usp=sharing
>(it's completely untested since i don't have a marshmallow device and
>my emulator isn't useable for this right now - so this is just for
>logging, the attempted fix will most likely fail).
>you could also try looking around /mnt and /storage with a terminal
>emulator as the root user. the point is to find the path to storage
>which the root user can access and write to.
>
>---
>Reply to this email directly or view it on GitHub:
>https://github.com/jensstein/oandbackup/issues/108#issuecomment-159064160

-- 
Sent from my Android device with K-9 Mail. Please excuse my brevity.
11-23 12:57:39.949 I/ActivityManager(5018): START u0 {act=android.intent.action.VIEW dat=file:///sdcard/oandbackups/oandbackup.log typ=text/plain cmp=android/com.android.internal.app.ResolverActivity} from uid 10059 on display 0
11-23 12:57:41.864 I/ActivityManager(5018): START u0 {act=android.intent.action.VIEW dat=file:///sdcard/oandbackups/oandbackup.log typ=text/plain flg=0x3000000 cmp=com.amaze.filemanager/.activities.TextReader} from uid 10059 on display 0
11-23 13:15:00.715 I/GeckoNotificationHelper(30187): Send {"id":"{5814479f-0a69-4836-9170-48086ef1cd44}","handlerKey":"downloads","cookie":"\"\/storage\/emulated\/0\/Download\/oandbackup-debug.apkhttps:\/\/doc-0g-3o-docs.googleusercontent.com\/docs\/securesc\/ha0ro937gcuc7l7deffksulhg5h7mbp1\/an2lentv6ndqql0cjv9orgmd3b31632g\/1448308800000\/10277192567216509076\/*\/0B0bNUNWd2wPPVVJ6SWd5UV9Rcms?e=downloadMon Nov 23 2015 13:15:00 GMT-0800 (PST)\"","eventType":"notification-closed"}
11-23 13:15:04.522 I/GeckoNotificationHelper(30187): Send {"id":"{9f8715ab-b1a4-4e55-bc9d-90c57d2c5126}","eventType":"notification-clicked","handlerKey":"downloads","cookie":"\"\/storage\/emulated\/0\/Download\/oandbackup-debug.apkhttps:\/\/doc-0g-3o-docs.googleusercontent.com\/docs\/securesc\/ha0ro937gcuc7l7deffksulhg5h7mbp1\/an2lentv6ndqql0cjv9orgmd3b31632g\/1448308800000\/10277192567216509076\/*\/0B0bNUNWd2wPPVVJ6SWd5UV9Rcms?e=downloadMon Nov 23 2015 13:15:00 GMT-0800 (PST)\""}
11-23 13:15:04.523 I/GeckoNotificationHelper(30187): Send {"id":"{9f8715ab-b1a4-4e55-bc9d-90c57d2c5126}","handlerKey":"downloads","cookie":"\"\/storage\/emulated\/0\/Download\/oandbackup-debug.apkhttps:\/\/doc-0g-3o-docs.googleusercontent.com\/docs\/securesc\/ha0ro937gcuc7l7deffksulhg5h7mbp1\/an2lentv6ndqql0cjv9orgmd3b31632g\/1448308800000\/10277192567216509076\/*\/0B0bNUNWd2wPPVVJ6SWd5UV9Rcms?e=downloadMon Nov 23 2015 13:15:00 GMT-0800 (PST)\"","eventType":"notification-closed"}
11-23 13:15:05.281 I/ActivityManager(5018): START u0 {act=android.intent.action.VIEW dat=file:///storage/emulated/0/Download/oandbackup-debug.apk typ=application/vnd.android.package-archive flg=0x4000000 cmp=com.android.packageinstaller/.PackageInstallerActivity (has extras)} from uid 10044 on display 0
11-23 13:15:06.570 I/ActivityManager(5018): START u0 {dat=file:///storage/emulated/0/Download/oandbackup-debug.apk cmp=com.android.packageinstaller/.InstallAppProgress (has extras)} from uid 10016 on display 0
11-23 13:15:06.639 D/DefContainer(22669): Copying /storage/emulated/0/Download/oandbackup-debug.apk to base.apk
11-23 13:15:13.097 I/GeckoNotificationHelper(30187): Send {"id":"{61b23950-d38d-4569-99fa-4bd41374ba54}","handlerKey":"downloads","cookie":"\"\/storage\/emulated\/0\/Download\/oandbackup-debug(2).apkhttps:\/\/doc-0g-3o-docs.googleusercontent.com\/docs\/securesc\/ha0ro937gcuc7l7deffksulhg5h7mbp1\/an2lentv6ndqql0cjv9orgmd3b31632g\/1448308800000\/10277192567216509076\/*\/0B0bNUNWd2wPPVVJ6SWd5UV9Rcms?e=downloadMon Nov 23 2015 13:15:13 GMT-0800 (PST)\"","eventType":"notification-closed"}
11-23 13:15:15.722 I/GeckoNotificationHelper(30187): Send {"id":"{f9389a8c-236b-4827-84a4-eb05e718bc19}","eventType":"notification-clicked","handlerKey":"downloads","cookie":"\"\/storage\/emulated\/0\/Download\/oandbackup-debug(2).apkhttps:\/\/doc-0g-3o-docs.googleusercontent.com\/docs\/securesc\/ha0ro937gcuc7l7deffksulhg5h7mbp1\/an2lentv6ndqql0cjv9orgmd3b31632g\/1448308800000\/10277192567216509076\/*\/0B0bNUNWd2wPPVVJ6SWd5UV9Rcms?e=downloadMon Nov 23 2015 13:15:13 GMT-0800 (PST)\""}
11-23 13:15:15.722 I/GeckoNotificationHelper(30187): Send {"id":"{f9389a8c-236b-4827-84a4-eb05e718bc19}","handlerKey":"downloads","cookie":"\"\/storage\/emulated\/0\/Download\/oandbackup-debug(2).apkhttps:\/\/doc-0g-3o-docs.googleusercontent.com\/docs\/securesc\/ha0ro937gcuc7l7deffksulhg5h7mbp1\/an2lentv6ndqql0cjv9orgmd3b31632g\/1448308800000\/10277192567216509076\/*\/0B0bNUNWd2wPPVVJ6SWd5UV9Rcms?e=downloadMon Nov 23 2015 13:15:13 GMT-0800 (PST)\"","eventType":"notification-closed"}
11-23 13:15:16.479 I/ActivityManager(5018): START u0 {act=android.intent.action.VIEW dat=file:///storage/emulated/0/Download/oandbackup-debug(2).apk typ=application/vnd.android.package-archive flg=0x4000000 cmp=com.android.packageinstaller/.PackageInstallerActivity (has extras)} from uid 10044 on display 0
11-23 13:15:19.916 I/ActivityManager(5018): START u0 {dat=file:///storage/emulated/0/Download/oandbackup-debug(2).apk cmp=com.android.packageinstaller/.InstallAppProgress (has extras)} from uid 10016 on display 0
11-23 13:15:20.003 D/DefContainer(22669): Copying /storage/emulated/0/Download/oandbackup-debug(2).apk to base.apk
11-23 13:15:58.430 I/ActivityManager(5018): START u0 {act=android.intent.action.VIEW dat=file:///sdcard/Download/oandbackup-debug.apk typ=application/vnd.android.package-archive cmp=com.android.packageinstaller/.PackageInstallerActivity} from uid 10059 on display 0
11-23 13:15:59.495 I/ActivityManager(5018): START u0 {dat=file:///sdcard/Download/oandbackup-debug.apk cmp=com.android.packageinstaller/.InstallAppProgress (has extras)} from uid 10016 on display 0
11-23 13:15:59.571 D/DefContainer(22669): Copying /sdcard/Download/oandbackup-debug.apk to base.apk
11-23 13:16:45.888 I/GeckoNotificationHelper(30187): Send {"id":"{1be403eb-64f5-4d93-b409-5c03a5c7d648}","handlerKey":"downloads","cookie":"\"\/storage\/emulated\/0\/Download\/oandbackup-debug-2.apkhttps:\/\/doc-14-3o-docs.googleusercontent.com\/docs\/securesc\/ha0ro937gcuc7l7deffksulhg5h7mbp1\/2otc5a4nrea2gbulig5ki3d9r0pp8lj7\/1448308800000\/10277192567216509076\/*\/0B0bNUNWd2wPPVEVlN0tuakpHV28?e=downloadMon Nov 23 2015 13:16:45 GMT-0800 (PST)\"","eventType":"notification-closed"}
11-23 13:16:51.720 I/GeckoNotificationHelper(30187): Send {"id":"{44a29b94-4e4e-4eb0-ad65-aaaf4e00bc58}","eventType":"notification-clicked","handlerKey":"downloads","cookie":"\"\/storage\/emulated\/0\/Download\/oandbackup-debug-2.apkhttps:\/\/doc-14-3o-docs.googleusercontent.com\/docs\/securesc\/ha0ro937gcuc7l7deffksulhg5h7mbp1\/2otc5a4nrea2gbulig5ki3d9r0pp8lj7\/1448308800000\/10277192567216509076\/*\/0B0bNUNWd2wPPVEVlN0tuakpHV28?e=downloadMon Nov 23 2015 13:16:45 GMT-0800 (PST)\""}
11-23 13:16:51.721 I/GeckoNotificationHelper(30187): Send {"id":"{44a29b94-4e4e-4eb0-ad65-aaaf4e00bc58}","handlerKey":"downloads","cookie":"\"\/storage\/emulated\/0\/Download\/oandbackup-debug-2.apkhttps:\/\/doc-14-3o-docs.googleusercontent.com\/docs\/securesc\/ha0ro937gcuc7l7deffksulhg5h7mbp1\/2otc5a4nrea2gbulig5ki3d9r0pp8lj7\/1448308800000\/10277192567216509076\/*\/0B0bNUNWd2wPPVEVlN0tuakpHV28?e=downloadMon Nov 23 2015 13:16:45 GMT-0800 (PST)\"","eventType":"notification-closed"}
11-23 13:16:52.509 I/ActivityManager(5018): START u0 {act=android.intent.action.VIEW dat=file:///storage/emulated/0/Download/oandbackup-debug-2.apk typ=application/vnd.android.package-archive flg=0x4000000 cmp=com.android.packageinstaller/.PackageInstallerActivity (has extras)} from uid 10044 on display 0
11-23 13:16:53.698 I/ActivityManager(5018): START u0 {dat=file:///storage/emulated/0/Download/oandbackup-debug-2.apk cmp=com.android.packageinstaller/.InstallAppProgress (has extras)} from uid 10016 on display 0
11-23 13:16:53.768 D/DefContainer(22669): Copying /storage/emulated/0/Download/oandbackup-debug-2.apk to base.apk
11-23 13:17:49.845 I/ActivityManager(5018): START u0 {act=android.intent.action.VIEW dat=file:///sdcard/Download/oandbackup-debug.apk typ=application/vnd.android.package-archive cmp=com.android.packageinstaller/.PackageInstallerActivity} from uid 10059 on display 0
11-23 13:17:50.942 I/ActivityManager(5018): START u0 {dat=file:///sdcard/Download/oandbackup-debug.apk cmp=com.android.packageinstaller/.InstallAppProgress (has extras)} from uid 10016 on display 0
11-23 13:17:51.022 D/DefContainer(22669): Copying /sdcard/Download/oandbackup-debug.apk to base.apk
11-23 13:17:54.190 I/ActivityManager(5018): START u0 {act=android.intent.action.MAIN cat=[android.intent.category.LAUNCHER] flg=0x10000000 pkg=dk.jens.backup cmp=dk.jens.backup/.OAndBackup} from uid 10016 on display 0
11-23 13:17:54.253 I/ActivityManager(5018): Start proc 24299:dk.jens.backup/u0a92 for activity dk.jens.backup/.OAndBackup
11-23 13:17:54.563 I/ActivityManager(5018): Displayed dk.jens.backup/.OAndBackup: +323ms
11-23 13:17:59.143 E/oandbackup(24299): oandbackup.log: error while reading logfile: org.json.JSONException: Value java.io.FileNotFoundException of type java.lang.String cannot be converted to JSONObject
11-23 13:17:59.144 E/oandbackup(24299): test: error while reading logfile: org.json.JSONException: Value java.io.FileNotFoundException of type java.lang.String cannot be converted to JSONObject
11-23 13:18:03.329 I/oandbackup(24299): backupdir: /storage/emulated/0/oandbackups/am.ed.exportcontacts
11-23 13:18:03.329 I/oandbackup(24299): backup: Export Contacts
11-23 13:18:03.690 I/oandbackup(24299): return: 1 / error

**Comment by takiainen** [comment]
*https://drive.google.com/file/d/0B0bNUNWd2wPPVEVlN0tuakpHV28/view?usp=sharing* 

I was able to restore my Lollipop backups into Marshmallow with the above version. All other versions gave the same error mentioned by wisemonkey. Saved my day, thank you!

**Comment by wisemonkey** [comment]
Just tried that version and it didn't give me an error. And it says backup complete is there a way to verify if backup is correct?

On November 23, 2015 4:18:15 PM PST, takiainen <notifications@github.com> wrote:
>*https://drive.google.com/file/d/0B0bNUNWd2wPPVEVlN0tuakpHV28/view?usp=sharing*
>
>
>I was able to restore my Lollipop backups into Marshmallow with the
>above version. All other versions gave the same error mentioned by
>wisemonkey. Saved my day, thank you!
>
>
>
>---
>Reply to this email directly or view it on GitHub:
>https://github.com/jensstein/oandbackup/issues/108#issuecomment-159111632

-- 
Sent from my Android device with K-9 Mail. Please excuse my brevity.

**Comment by jensstein** [comment]
at the moment there isn't any way to check the integrety of the backups (i plan to check md5sums in the future). but you can either try unzipping the backed-up files - apks can be unzipped too - or you can try restoring a non-essential app to test it.
hopefully @takiainen's post means that it works in general.

**Comment by malaverdiere** [comment]
I tried the special build linked here and had no success. I cannot restore. I get `no apk to install:... cannot restore data without restoring apk, package is not installed`

Log entries:
    2016/01/14 - 16:43:22: no apk to install: acr.browser.lightning [acr.browser.lightning]
    2016/01/14 - 16:43:22: cannot restore data without restoring apk, package is not installed [acr.browser.lightning]
    2016/01/14 - 16:43:22: no apk to install: at.bitfire.davdroid [at.bitfire.davdroid]
    2016/01/14 - 16:43:22: cannot restore data without restoring apk, package is not installed [at.bitfire.davdroid]
    2016/01/14 - 16:43:22: no apk to install: com.android.keepass [com.android.keepass]
    2016/01/14 - 16:43:22: cannot restore data without restoring apk, package is not installed [com.android.keepass]
    2016/01/14 - 16:43:22: no apk to install: com.fsck.k9 [com.fsck.k9]
    2016/01/14 - 16:43:22: cannot restore data without restoring apk, package is not installed [com.fsck.k9]
    2016/01/14 - 16:43:22: no apk to install: com.haringeymobile.ukweather [com.haringeymobile.ukweather]
    2016/01/14 - 16:43:22: cannot restore data without restoring apk, package is not installed [com.haringeymobile.ukweather]
    2016/01/14 - 16:43:22: no apk to install: com.kpz.pomodorotasks.activity [com.kpz.pomodorotasks.activity]
    2016/01/14 - 16:43:22: cannot restore data without restoring apk, package is not installed [com.kpz.pomodorotasks.activity]
    2016/01/14 - 16:43:22: no apk to install: com.totsp.bookworm [com.totsp.bookworm]
    2016/01/14 - 16:43:22: cannot restore data without restoring apk, package is not installed [com.totsp.bookworm]
    2016/01/14 - 16:43:22: no apk to install: de.arnowelzel.android.periodical [de.arnowelzel.android.periodical]
    2016/01/14 - 16:43:22: cannot restore data without restoring apk, package is not installed [de.arnowelzel.android.periodical]
    2016/01/14 - 16:43:22: no apk to install: de.luhmer.owncloudnewsreader [de.luhmer.owncloudnewsreader]
    2016/01/14 - 16:43:22: cannot restore data without restoring apk, package is not installed [de.luhmer.owncloudnewsreader]
    2016/01/14 - 16:43:22: no apk to install: dk.jens.backup [dk.jens.backup]
    2016/01/14 - 16:43:22: cannot restore data without restoring apk, package is not installed [dk.jens.backup]
    2016/01/14 - 16:43:22: no apk to install: net.bible.android.activity [net.bible.android.activity]
    2016/01/14 - 16:43:22: cannot restore data without restoring apk, package is not installed [net.bible.android.activity]
    2016/01/14 - 16:43:22: no apk to install: org.aykit.MyOwnNotes [org.aykit.MyOwnNotes]
    2016/01/14 - 16:43:22: cannot restore data without restoring apk, package is not installed [org.aykit.MyOwnNotes]
    2016/01/14 - 16:43:22: no apk to install: org.coolreader [org.coolreader]
    2016/01/14 - 16:43:22: cannot restore data without restoring apk, package is not installed [org.coolreader]
    2016/01/14 - 16:43:22: no apk to install: org.fdroid.fdroid [org.fdroid.fdroid]
    2016/01/14 - 16:43:22: cannot restore data without restoring apk, package is not installed [org.fdroid.fdroid]
    2016/01/14 - 16:43:22: no apk to install: org.fedorahosted.freeotp [org.fedorahosted.freeotp]
    2016/01/14 - 16:43:22: cannot restore data without restoring apk, package is not installed [org.fedorahosted.freeotp]
    2016/01/14 - 16:43:22: no apk to install: org.schabi.newpipe [org.schabi.newpipe]
    2016/01/14 - 16:43:22: cannot restore data without restoring apk, package is not installed [org.schabi.newpipe]
    2016/01/14 - 16:43:22: no apk to install: org.scoutant.blokish [org.scoutant.blokish]
    2016/01/14 - 16:43:22: cannot restore data without restoring apk, package is not installed [org.scoutant.blokish]
    2016/01/14 - 16:43:22: no apk to install: org.sufficientlysecure.viewer [org.sufficientlysecure.viewer]
    2016/01/14 - 16:43:22: cannot restore data without restoring apk, package is not installed [org.sufficientlysecure.viewer]
    2016/01/14 - 16:43:22: no apk to install: org.telegram.messenger [org.telegram.messenger]
    2016/01/14 - 16:43:22: cannot restore data without restoring apk, package is not installed [org.telegram.messenger]
    2016/01/14 - 16:43:22: no apk to install: org.zephyrsoft.trackworktime [org.zephyrsoft.trackworktime]
    2016/01/14 - 16:43:22: cannot restore data without restoring apk, package is not installed [org.zephyrsoft.trackworktime]
    2016/01/14 - 16:43:22: no apk to install: ru.o2genum.coregame [ru.o2genum.coregame]
    2016/01/14 - 16:43:22: cannot restore data without restoring apk, package is not installed [ru.o2genum.coregame]
    2016/01/14 - 16:44:17: no apk to install: de.arnowelzel.android.periodical [de.arnowelzel.android.periodical]
    2016/01/14 - 16:44:17: cannot restore data without restoring apk, package is not installed [de.arnowelzel.android.periodical]
    2016/01/14 - 16:46:52: no apk to install: acr.browser.lightning [acr.browser.lightning]
    2016/01/14 - 16:46:52: cannot restore data without restoring apk, package is not installed [acr.browser.lightning]

**Comment by jensstein** [comment]
@malaverdiere this is likely a different problem so i could i get you to open a new issue?
and could i get you to look in the backup log for one of the apps that fail to restore (the text file in the backup subdirectory for the app, e.g. acr.browser.lightning.log) - there should be an entry called sourceDir. please give me the value of that entry and compare it to the filename of the backed up apk.
also, please try a restore with this version and give me a logcat: https://drive.google.com/file/d/0B0bNUNWd2wPPeHhGc0Vnb2lEQmM/view?usp=sharing
did you do both backup and restore on android 6 or was the backups from an earlier version?

**Comment by 78c59635bdd8** [comment]
I also installed the special build linked here over a month ago.  This worked for making my backups work.  It also works for restoring the apk.  However, when I encounter two problems when I attempt to restore data:
1. If I restore data to an apk, the phone says that the data was restored successfully, but it isn't restored at all. For example, say I restore an apk only -- so it is currently unconfigured.  Then I restore the data for that apk.  When I open the app, I would expect it to now be configured, but nothing changes.  It still shows as unconfigured.

2. If I restore the apk+data at the same time, such as K9 Mail, when I then try to open up the app, I get "Unfortunately, K-9 Mail has stopped".  The only way to fix this is to either a) clear the data from the app, or b) uninstall and reinstall the apk only, which of course forces you to reconfigure your app and lost your data.

**Comment by jensstein** [comment]
@78c59635bdd8 have you tried the apk linked above (https://drive.google.com/file/d/0B0bNUNWd2wPPeHhGc0Vnb2lEQmM/view?usp=sharing)?

**Comment by 78c59635bdd8** [comment]
That was it! Thanks! I had an older oandbackup-debug.apk.  The version you just linked to is working great! Sorry for the noise.  Thanks again for a great app!

**Comment by malaverdiere** [comment]
Sadly, it looks like my backups vanished. Is there any code path that forces a directory to be deleted specific in that build?

**Comment by jensstein** [comment]
no, there shouldn't be. but it's hard to tell what caused the bug since there was probably code in that apk which wasn't committed.
so all i can suggest unfortunately is that you try again with the version linked above.
when you say your backups vanished, do you mean that neither apks nor data are in the backup directory?

**Comment by malaverdiere** [comment]
Neither APKs nor data are in the backup directory - only oandbackups/oandbackup.log - which perfectly explains the error messages I shared.

**Comment by jensstein** [comment]
@malaverdiere yes, that does explain the error. it's hard to tell whether the bug was only present in the apk you used, so please be very cautious when you are using the new version and check that everything is actually backed up as expected since the bug might be related to your device or rom.
